src/alert.js

```diff
@@ -1,25 +1,25 @@
 import React, { forwardRef } from 'react'
 import PropTypes from 'prop-types'
+
+import { makeStyles } from '@material-ui/core/styles'
 import MUIAlert from '@material-ui/lab/Alert'
 import AlertTitle from '@material-ui/lab/AlertTitle'
 
-const Alert = forwardRef(({ severity, message, header, width, ...props }, ref) => {
-  const dimensions = { height: 'auto', width }
+const useStyles = makeStyles({ root: ({ width = '100%' }) => ({ height: 'auto', width }) })
```

Comment:
Do we need a default parameter for width when PropTypes exists?

Reply:
at runtime (in production), most React projects strips out `propTypes` altogether for performance measures.

Reply:
> what is root: ({ width = '100%' }) for? before setting it to { height: 'auto', width }

@kc-leung as @DoParkEQ mentioned, it's default parameter in combination with the destructuring feature of ES6 to emulate "named-parameter" behaviours that's not a part of Javascript. For instance:

```js
function dlog({ message = 'Generic message' } = {}) {
  console.log(message)
}

dlog() // logs "Generic message"
dlog({}) // logs "Generic message"
dlog({ message: 'Explicit message' }) // logs "Explicit message"
```